thread-manager/src/lib.rs

```diff
@@ -0,0 +1,380 @@
+use {
+    anyhow::Ok,
+    serde::{Deserialize, Serialize},
+    std::{collections::HashMap, ops::Deref, sync::Arc},
+};
+
+pub mod native_thread_runtime;
+pub mod policy;
+pub mod rayon_runtime;
+pub mod tokio_runtime;
+
+pub use {
+    native_thread_runtime::{JoinHandle, NativeConfig, NativeThreadRuntime},
+    policy::CoreAllocation,
+    rayon_runtime::{RayonConfig, RayonRuntime},
+    tokio_runtime::{TokioConfig, TokioRuntime},
+};
+pub type ConstString = Box<str>;
```

Comment:
Why do we want boxed strings? it seems like this is the whole cause of us not being able to simply clone the maps, but doesn't seem necessary to me at first glance.

Reply:
To make sure noone gets ideas about mutating them. But I guess they can be just String if that is somehow better. We would not be able to just clone the maps anyway, but I guess it would make code slightly cleaner. I'll check.

Reply:
Box's don't prevent mutation though. Just provide unique ownership

Reply:
Yeah I think unless we have a good reason to use box we shouldn't

Reply:
Done!